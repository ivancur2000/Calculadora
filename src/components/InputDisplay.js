import React from 'react';
import PropTypes from 'prop-types';

export const InputDisplay = ({ value }) =>{
    return(
        <input
            type="text"
            className='form-control'
            value={value.join('')}
            disabled
        />
    );
}
InputDisplay.propTypes = {
    value: PropTypes.array.isRequired,
}