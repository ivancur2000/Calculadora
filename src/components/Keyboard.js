import React from 'react';
import PropTypes from 'prop-types';

export const Keyboard = ({ handleOnClickNumber, handleOnClickOperation, handleOnReset, handleOnResolve })=>{

    const btnNumber = [1,2,3,4,5,6,7,8,9,0];
    const operations = ['+', '-', '/', '*'];

    return(
    <div className='flex flex-wrap'>
        <div className='row w-100'>
            <div className='col-6 number__pad'>
                {
                    btnNumber.map(number =>(
                        <button
                            className='btn btn-lg btn-success'
                            key={number}
                            onClick={ ()=> handleOnClickNumber( number ) }
                        >
                            {number}
                        </button>
                    ))
                }
            </div>
            <div className='col-6 operation__pad'>
                <button 
                    className='btn btn-lg btn-warning'
                    onClick={ handleOnReset }
                >
                    CE
                </button>
                <button 
                    className='btn btn-lg btn-danger'
                    onClick={ handleOnResolve }
                >
                    =
                </button>
                {
                    operations.map(symbol => (
                        <button
                            className='btn btn-lg btn-light'
                            key={symbol}
                            onClick={ ()=> handleOnClickOperation(symbol) }
                        >
                            {symbol}
                        </button>
                    ))
                }
            </div>
        </div>
    </div>
    );
}
Keyboard.propTypes = {
    handleOnClickNumber: PropTypes.func.isRequired,
    handleOnClickOperation: PropTypes.func.isRequired,
    handleOnReset: PropTypes.func.isRequired,
    handleOnResolve: PropTypes.func.isRequired,
}