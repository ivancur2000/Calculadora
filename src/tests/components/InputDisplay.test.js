import { shallow } from "enzyme";
import { InputDisplay } from "../../components/InputDisplay";

describe('Pruebas en el componente <InputDisplay />', () => {
    const value = ['1','2','3'];
    const wrapper = shallow(<InputDisplay value={value} />);

    test('Debe renderizar <InputDisplay /> correctamente', () => {
       expect(wrapper).toMatchSnapshot();
    });
    
    test('Debe mostrar el valor de array correctamente', ()=>{
        const valueString = value.join('');
        const input = wrapper.find('input');

        expect(input.prop('value')).toBe(valueString);
    });
});
