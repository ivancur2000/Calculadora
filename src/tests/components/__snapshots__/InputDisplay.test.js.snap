// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <InputDisplay /> Debe renderizar <InputDisplay /> correctamente 1`] = `
<input
  className="form-control"
  disabled={true}
  type="text"
  value="123"
/>
`;
